pydoctor/sphinx_ext/build_apidocs.py

```diff
@@ -76,14 +89,55 @@ def _run_pydoctor(name: str, arguments: Iterable[str], placeholders: Mapping[str
             logger.warning(line)
 
 
+def update_intersphinx_mapping(app: Sphinx, config: Config) -> None:
+    """
+    Called to build and inject our own intersphinx inventory object.
+    """
+    rtd_version = 'latest'
+    if os.environ.get('READTHEDOCS', '') == 'True':
+        rtd_version = os.environ.get('READTHEDOCS_VERSION', 'latest')
+
+    output_dir = pathlib.Path(app.outdir)
+
+    intersphinx_mapping = config.intersphinx_mapping
+    for key, value in config.pydoctor_intersphinx_mapping.items():
+        url, target_path = value
+
+        url = url.format(**{'rtd_version': rtd_version})
+        inventory_path = output_dir / target_path
+
+        intersphinx_mapping[key] = (url, str(inventory_path))
+
+    if not config.pydoctor_args:
+        raise ConfigError("Missing 'pydoctor_args'.")
+
+    placeholders = {
+        'outdir': app.outdir,
+        }
+
+    runs = config.pydoctor_args
+
+    if not isinstance(runs, Mapping):
+        # We have a single pydoctor call
+        runs = {'main': runs}
+
+    for key, value in runs.items():
+        _run_pydoctor(key, ['--make-intersphinx'] + value, placeholders)
+
+
 def setup(app: Sphinx) ->  Mapping[str, Any]:
     """
     Called by Sphinx when the extension is initialized.
 
     @return: The extension version and runtime options.
     """
-    app.connect('build-finished', on_build_finished)
     app.add_config_value("pydoctor_args", None, "env")
+    app.add_config_value("pydoctor_intersphinx_mapping", {}, "env")
+
+    # Make sure we have a lower priority than intersphinx extension.
+    app.connect('config-inited', update_intersphinx_mapping, priority=799)
```

Comment:
Is the intersphinx extension at priority 800? If so, maybe use 790 or something, just in case there is ever an extension that wants to go between `build_apidocs` and intersphinx.

Reply:
True. Thanks!